tratto/src/main/java/star/tratto/util/javaparser/JavaParserUtils.java

```diff
@@ -0,0 +1,1078 @@
+        for (Node comment: jpCallable.getAllContainedComments()) {
+            methodSignature = methodSignature.replace(comment.toString(), "");
+        }
+        // remove Javadoc comment
+        if (jpCallable.getComment().isPresent()) {
+            methodSignature = methodSignature.replaceAll("[\\s\\S]*\n", "");
+        }
+        // remove special characters
+        methodSignature = methodSignature.replaceAll("/\\*\\*([\\s\\S]*?)\\*/(\\n|\\r|\\t)*", "");
+        return methodSignature.trim().replaceAll(";$", "");
+    }
+
+    /**
+     * Returns the signature of a JavaParser method declaration.
+     *
+     * @param methodDeclaration a JavaParser method declaration
+     * @return a string representation of the signature. Signature follows the
+     * format:
+     *  "[modifiers] [type] [methodName]([parameters]) throws [exceptions]"
+     */
+    public static String getMethodSignature(MethodDeclaration methodDeclaration) {
+        String method = methodDeclaration.toString();
+        if (methodDeclaration.getBody().isPresent()) {
+            // Remove body
+            method = method.replace(methodDeclaration.getBody().get().toString(), "");
+        }
+        for (Node comment: methodDeclaration.getAllContainedComments()) {
+            // Remove comments within method signature
+            method = method.replace(comment.toString(), "");
+        }
+        if (methodDeclaration.getComment().isPresent()) {
+            // At this point, last line is method signature. Remove everything before that
+            method = method.replaceAll("[\\s\\S]*\n", "");
+        }
+        return method.trim().replaceAll(";$", "");
+    }
+
+    /**
+     * Gets all type parameters of a given method.
+     */
+    private static List<String> getTypeParameters(MethodUsage methodUsage) {
+        return methodUsage.getDeclaration().getTypeParameters()
+                .stream()
+                .map(ResolvedTypeParameterDeclaration::getName)
+                .toList();
+    }
+
+    /**
+     * Gets all formal parameters in the method definition. This method gets
+     * the type of each parameter, followed by an artificial name. For
+     * example,
+     *     "MethodUsage[get(int i)]"    => "List.of("int arg0")"
+     */
+    private static List<String> getParameters(MethodUsage methodUsage) {
+        ResolvedMethodDeclaration methodDeclaration = methodUsage.getDeclaration();
+        // iterate through each parameter in the method declaration.
+        List<String> methodParameters = new ArrayList<>();
+        for (int i = 0; i < methodDeclaration.getNumberOfParams(); i++) {
+            methodParameters.add(getTypeWithoutPackages(methodDeclaration.getParam(i).getType()) + " arg" + i);
+        }
+        return methodParameters;
+    }
+
+    /**
+     * Gets the types of all exceptions that can be thrown by a given method.
+     */
+    private static List<String> getExceptions(MethodUsage methodUsage) {
```

Comment:
Why does this return a string?  That seems to be giving up information.  Also, what format is the string?  Fully qualified name?  Binary name?  Something else?  And why are they returned without packages?